src/functions.php

```diff
@@ -42,6 +43,39 @@
 use function reset;
 use function substr;
 
+/**
+ * Check whether all servers support executing a write stage on a secondary.
+ *
+ * @internal
+ * @see https://github.com/mongodb/specifications/blob/master/source/crud/crud.rst#aggregation-pipelines-with-write-stages
+ */
+function all_servers_support_write_stage_on_secondary(Manager $manager): bool
+{
+    static $wireVersionForWriteStageOnSecondary = 13;
+
+    $servers = $manager->getServers();
```

Comment:
[`Manager::getServers()`](https://github.com/mongodb/mongo-php-driver/blob/1.11.1/src/MongoDB/Manager.c#L671) calls [`mongoc_client_get_server_descriptions`](http://mongoc.org/libmongoc/current/mongoc_client_get_server_descriptions.html), which doesn't initialize the topology. The PHP driver does model all of the various server types reported by libmongoc, so unknown (and other non-usable) servers will get reported here.

I don't believe we need to worry about a case where SDAM is initialized but there are no servers in the topology. The spec describes that as a [terminal condition](https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#other-topologytypes), after which the driver would be left in an unusable state.